code/_onclick/hud/action.dm

```diff
@@ -288,6 +291,27 @@
 	to_chat(usr, SPAN_NOTICE("You press the button on the exterior of \the [target_clothing]."))
 	target_clothing.action_circuit.activate_pin(1)
 
+/datum/action/eye
+	action_type = AB_GENERIC
+	check_flags = AB_CHECK_LYING|AB_CHECK_STUNNED
+	var/eye_type = /mob/abstract/eye
+	var/target_type = PARENT_TARGET //The relevant owner of the proc to be called by the action.
+
+/datum/action/eye/New(var/datum/component/eye/eye_component)
+	switch(target_type)
```

Comment:
Verify that the eye_component is of appropriate type on creation before going forward, with !istype() -> crashwith